File: app/src/main/java/com/fos/fosmvp/ui/login/presenter/LoginPresenter.java
package com.fos.fosmvp.ui.login.presenter;


import com.fos.fosmvp.common.base.BaseApplication;
import com.fos.fosmvp.common.base.BaseResponse;
import com.fos.fosmvp.common.baserx.RxObserver;
import com.fos.fosmvp.entity.login.UserEntity;
import com.fos.fosmvp.common.http.Api;
import com.fos.fosmvp.ui.login.contract.LoginContract;

import java.util.HashMap;
import java.util.Map;

import io.reactivex.Observable;

public class LoginPresenter extends LoginContract.Presenter {
    @Override
    public void getLoginRequest(String tel, String password) {
        Map<String, Object> argMap = new HashMap<>();
        argMap.put("name", tel);
        argMap.put("password", password);
        Observable<BaseResponse<UserEntity>> observable = Api.initObservable(getMModel().getLoginData(argMap));
        RxObserver rxObserver = new RxObserver<BaseResponse<UserEntity>>(BaseApplication.Companion.getAppContext()) {
            @Override
            protected void _onNext(BaseResponse<UserEntity> res) {
                if (res.isSucceed()) {
                    getMView().returnLoginSucceed(res.getData());
                } else {
                    getMView().returnLoginFail(res, false);
                }
            }

            @Override
            protected void _onError(String message) {
                getMView().returnLoginFail(null, true);
            }
        };
        observable.subscribe(rxObserver);
        getMRxManage().add(rxObserver);
    }


}

File: app/src/main/java/com/fos/fosmvp/ui/login/presenter/LoginPresenter.kt
package com.fos.fosmvp.ui.login.presenter


import com.fos.fosmvp.common.base.BaseApplication
import com.fos.fosmvp.common.base.BaseResponse
import com.fos.fosmvp.common.baserx.RxObserver
import com.fos.fosmvp.common.http.Api
import com.fos.fosmvp.entity.login.UserEntity
import com.fos.fosmvp.ui.login.contract.LoginContract
import io.reactivex.Observable


class LoginPresenter : LoginContract.Presenter() {
    override fun getLoginRequest(tel: String, password: String) {
        val argMap = HashMap<String, String>()
        argMap.put("name", tel)
        argMap.put("password", password)
        var observable : Observable<BaseResponse<UserEntity>> = Api.initObservable(mModel!!.getLoginData(argMap)) as Observable<BaseResponse<UserEntity>>
        var rxObserver = object : RxObserver<BaseResponse<UserEntity>>(BaseApplication.appContext!!) {
            override fun _onNext(res: BaseResponse<UserEntity>) {
                if (res.isSucceed) {
                    mView!!.returnLoginSucceed(res.data!!)
                } else {
                    mView!!.returnLoginFail(res, false)
                }
            }

            override fun _onError(message: String) {
                mView!!.returnLoginFail(null, true)
            }
        }
        observable.subscribe(rxObserver)
        mRxManage.add(rxObserver)
    }


}
